src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -1174,13 +1172,24 @@ private static Property GetActualProperty(Property property, Class c)
         private void GenerateFieldGetter(Field field, Class @class, QualifiedType returnType)
         {
             var name = @class.Layout.Fields.First(f => f.FieldPtr == field.OriginalPtr).Name;
+            String returnVar;
+            if (@class.IsValueType)
+                returnVar = $@"{Helpers.InstanceField}.{SafeIdentifier(name)}";
+            else
+            {
+                returnVar = $@"(({TypePrinter.PrintNative(@class)}*) {Helpers.InstanceIdentifier})->{SafeIdentifier(name)}";
+                // Class field getter should return a reference object instead of a copy. Wrapping `returnVar` in
+                // IntPtr ensures that non-copying object constructor is invoked.
+                Class typeClass;
+                if (field.Type.TryGetClass(out typeClass) && !typeClass.IsValueType)
+                    returnVar = $"new global::System.IntPtr(&{returnVar})";
```

Comment:
One last nit here that I remembered: You should use `CSharpTypePrinter.IntPtrType` here instead of hardcoding the type.